### ISSUE TITLE
Object trainig

### ISSUE THREAD

**Issue by melikavhf** [DavidTorresOcana/ros_object_detector#2]
hello . how can I train my own object and how to replace in this package?
how should I create my own model? Is there any package to create these files? 
 sorry I'm not familiar with object detection .

**Comment by DavidTorresOcana** [comment]
Hello

I recommend you to explore/use Tensorflow's Object Detection libraries/code: https://github.com/tensorflow/models/tree/master/research/object_detection

With that, you should be able to train models on your data, freeze your model and include it into this ROS package.
Once you have your model trained and [frozen ](https://blog.metaflow.fr/tensorflow-how-to-freeze-a-model-and-serve-it-with-a-python-api-d4f3596b3adc) , place it inside [/include/ros_object_detector](https://github.com/DavidTorresOcana/ros_object_detector/tree/master/include/ros_object_detector) and run the node with the appropriate parameters.
You will need a Tensorflow [frozen ](https://blog.metaflow.fr/tensorflow-how-to-freeze-a-model-and-serve-it-with-a-python-api-d4f3596b3adc) graph on .pb format and the labels definition into .pbtxt format.

Please explore  [/include/ros_object_detector](https://github.com/DavidTorresOcana/ros_object_detector/tree/master/include/ros_object_detector) to see how the "pothole_detector" files estructure is.

Object detection (with Convolutional Neural Networks in this case) is Machine Learning so, if you are not familiar with Machine Learning, I recommend you this [course  ](https://www.coursera.org/specializations/deep-learning)and take a look to [Tensorflow tutorials](https://www.tensorflow.org/tutorials/) (if you were not familiar with it) 

Regards

**Comment by melikavhf** [comment]
Thanks for attention . how can I create .pbtxt file and .pb? and how should I connect these file to each other?

**Comment by melikavhf** [comment]
and do I need special dataset? or no ? just .pb ?

**Comment by DavidTorresOcana** [comment]
Hello

With all the respect, this is not related with any issue of the repo so so we need to closing it. Feel free to contact me personally on email.
Please take a look to the references above to learn more about machine learning.
If you were looking for a general object detector, [Yolov2 ](https://pjreddie.com/darknet/yolo/)is coming to this repo soon, so you may only need to wait.

Regards
